### ISSUE TITLE
static properties

### ISSUE THREAD

**Issue by graingert** [Shopify/eslint-plugin-shopify#66]
# Prefer class properties to assignment of literals to the class. (prefer-class-properties)

Static class properties allow you to set initial values for class properties, instead of setting these after the declaration of a class.

## Rule Details

This rule takes one argument. If it is `'always'` (default) then it warns when it finds an assignment to `this` in a class constructor that could instead be a class property. If set to `'never'`, it will warn when any class properties are found.

The following assignments are all considered warnings when using the default or explicitly setting the argument to `'always'`:

```js
class Foo {
}

    Foo.bar = 'bar';
    Foo.baz = 123;
    Foo.qux = {
      foo: 'bar',
      baz: [1, 2, 3];
    };
```

The following are considered warnings when setting the argument to `'never'`:

```js
class Foo {
  static baz = 123;
  static qux = something();
}
```

## When Not To Use It

If you do not care how literal class members are assigned, you can safely disable this rule.

## Further Reading

- [Class properties proposal](https://github.com/jeffmo/es-class-fields-and-static-properties)

**Comment by alexandcote** [comment]
First, thank you for the proposition. Unfortunately, no one had time to look into this since you open the issue 2 years ago. This package moved to [`web-foundation`](https://github.com/shopify/web-foundation). Feel free to reopen this issue if you think it's still relevant.
